src/Filter/ORM/AbstractFilter.php

```diff
@@ -34,10 +35,15 @@ public function getFilterManager()
 
     protected function typeCastField($metadata, $field, $value, $format, $doNotTypecastDatetime = false)
     {
-        if (! isset($metadata->fieldMappings[$field])) {
+        // If association just return
+        if (isset($metadata->associationMappings[$field])) {
             return $value;
         }
 
+        if (! isset($metadata->fieldMappings[$field])) {
+            throw new InvalidFilterException('Field [' . $field . '] does not exist on target entity.');
```

Comment:
Please use `sprintf` and quotes `"` for field name (instead of `[]`), remove also `.` at the end - I can't see it with other messages in this PR.